0006/README.md

```diff
@@ -0,0 +1,49 @@
+---
+id: 0006
+title: Resiliency in Tinkerbell Provisioner.
+status: Ideation
+authors: Chitrabasu Khare <chitrabasukhare89@gmail.com>
+---
+
+## Summary:
+
+This RFD proposes to have a resilient framework for Tinkerbell Provisioner.
+
+
+## Goals:
+
+- Bring resiliency in Tinkerbell. 
+- Tinkerbell as a setup should be highly available.
+
+## Content: 
+
+- We would like to have a framework where Tinkerbell as a solution is fault-tolerant.
+- All most all of the core services are managed by provisioner. So, it will be good to have a framework where provisioner is highly available.
+- It can be a multinode provisioner solution with active-active or active-passive implementation.
+- The Provisioner services should be partition tolerant during network failures or communication failure between provisioner nodes.
+- Additional logic would be required to sync services running inside provisioner nodes.
+- A separate machine to place an L3/L7 Load balancer or proxy would be required.
```

Comment:
This is something that can be done today and it does not require any work. Vagrant setup is just a way to setup Tinkerbell, the same as the docker-compose. This to me looks more like something that has to be developed in the documentation and as ops experience.

Reply:
My suggestion is to have this feature for an active-active multi-node provisioner. In that case request needs to be routed to one of the node, hence we need a LB. It should be independent of the `type of deployment` of setup.

Reply:
As a blog, guide, or documentation, I could see several HA avenues being described:
* part 1: deploying with kubernetes and microservices
* part 2: local loadbalancers and db replication
* part 3: cloud loadbalancers and cloud sql

Tinkerbell should work for users wanting to take any of these solutions, and perhaps many more. I agree with @gianarb that the topic of this proposal is best explored outside of the proposal system.  The shortcomings or gaps that are found in the process (why couldn't tink run in kubernetes, why couldn't the database be configured to replicate, how did loadbalanced API servers miscommunicate) would make for more concise proposals.